### DISCUSSION TITLE
OVER clause

### DISCUSSION THREAD

**Discussion by tlackemann** [planetscale/discussion#290] [Q&A]
We have recently switched over to PlanetScale and noticed an issue with some commands that I believe should work based on MySQL comparability.

```sql
select x, sum(id_count) over (order by x) as y from (select DATE(table.created_at) as x, count(distinct (table.id)) as id_count from table where table.theme_id in (1, 2) group by x) data order by x;
```

Results in `syntax error at position 29 near 'over'` whereas it returns as expected using plain MySQL.

Based on some other answers in these discussions, `OVER` should be supported. Is there something unique we're doing in the above query that would result in this not being supported?

**Comment by orware** [comment]
Hello @tlackemann!

That is interesting and could definitely be a result of some missing support in our Vitess layer as I've found a few related discussions about the topic [over there](https://github.com/vitessio/vitess/issues/4099), even though I was able to make use of the `ROW_NUMBER()` function and the `OVER()` clause recently when responding to [discussion#287](https://github.com/planetscale/discussion/discussions/287#discussioncomment-3703201) which you probably saw before writing up your post above. 

So it may still be something we may need to include in our [MySQL compatibility](https://planetscale.com/docs/reference/mysql-compatibility) page perhaps, although work towards closing the compatibility gap is constantly occurring.

In attempting to run a similar test on my end, I was able to confirm similar behavior:

```sql
select x, 
       sum(id_count) over (order by x) as y 
from (
    select DATE(table.created_at) as x, 
              count(distinct (table.id)) as id_count 
    from table 
    where table.theme_id in (1, 2) 
    group by x
) data 
order by x;
```

I do wonder however if adjusting your query to the following would be equivalent for your particular use case in this situation?

```sql
select x, 
       sum(id_count) as y 
from (
    select DATE(table.created_at) as x, 
              count(distinct (table.id)) as id_count 
    from table 
    where table.theme_id in (1, 2) 
    group by x
) data 
group by x
order by x;
```

If you can give that a test to compare results I'd be curious to know if that adjustment could work.
